src/create.js

```diff
@@ -73,14 +77,26 @@ module.exports = function createDriftwood (primaryLogger) {
     globalState.loggers.push(log)
     return log
 
+    function intercept (args) {
+      if (interceptors && interceptors.length > 0) {
+        for (var i = 0; i < interceptors.length; i++) {
+          args = interceptors[i](args.slice())
+        }
+      }
+      return args
+    }
+
     function createAPI () {
       _.each(LEVELS.NAMES, function addLevelLogger (logLevel) {
         var index = LEVELS.INDEX[logLevel]
         log[logLevel] = state.enabled
           ? function levelLogger () {
             if (index >= LEVELS.INDEX[state.level]) {
               try {
-                logger(name, logLevel, new Date(), argsToComponents(arguments))
+                var args = [].slice.apply(arguments)
+                args = intercept(args)
```

Comment:
Would it not be nicer to give the interceptor the components? Also would it be useful to give the interceptor the logger name, log level and the date?

Reply:
The components seemed to me an internal construct, not something to be exposed in the public API. Doesn't passing the plain arguments seem a lot more intuitive and straight-forward for the average user? It would also spare the library from having to validate the structure of the return value.

Reply:
The components is what additional loggers get given as well - it isn't an internal construct. The nice thing about components is that all the logic of working out what the user as tried to log is done for you.

Reply:
At the cost of making interceptors more complex. But fair enough.

Reply:
Why does it make them more complex?

Reply:
They'll have to expect and return objects with specific structure rather than just the original arguments.

Reply:
But they'd have to expect and return the structure of the arguments if they don't. You can provide various types of data to driftwood logging, including objects and error instances. That is why we have the destructured version of the provided arguments, to make it easier to work with.

Reply:
I have the feeling that, for the average use-case of interceptors, components won't be easier to work with. But it's just a feeling. And it's best to remain consistent with additional loggers, as you point out.